### PR TITLE
Creating a Simple Chat App in React.js using the Chat Engine API

### PR DESCRIPTION
**Pull request by Neema-2016**
This fixes issue #3682 
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+title: Creating a Simple Chat App in React.js using the Chat Engine API
+---
+
+
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:5]: Remove these spaces where necessary in the whole article.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+Ever thought of coming up with a chat application, but going the Node.js and or the socket.io way seemed pretty cumbersome for someone solely equipped with front-end skills like React.js? 
+Well, this article fits you right. 
+The reader is brought to an understanding of the Chat engine API that we will be using, to come up with a precise yet functional Direct-Messaging application built in React.js. 
+Approximating to other APIs that deliver to the system, responses from users, and sends back responses of the system to the user, the Chat engine API does this also only with better functionalities.
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:9]: Try breaking down your paragraphs

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+<!--more-->
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:11]: Why do we have this here?

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
+---
+title: Creating a Simple Chat App in React.js using the Chat Engine API
+---
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:3]: Dow we need this part here?

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+<!--more-->
+This tool saves you a lot of implementation time you would otherwise need on many lines of code.
+Chat Engine API can implement; 
+ - A group chat application.
+ - Direct messages chat application that would work like Facebook messenger .
+
+### Prerequisites
+You will need to be equipped with what follows next before jumping right to it.
+1. Primary React.js library mastery. Head to [the official documentation](https://reactjs.org/tutorial/tutorial.html) if there's a need to familiarize yourself with React.js.
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:19]: This is quite wordy, simplify your sentences where applicable throughout your article.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+2. Installed on your system Node.js, which you could access from [here](https://nodejs.org/en/download/). 
+3. [Node Package Manager (NPM)](https://docs.npmjs.com/cli/v6/commands/npm) know-how.
+
+### What Chat Engine API is?
+The above introduction gave us an overview of this API and we realized how the API could be consumed to build easily; chat services.
+It also allows the hosting of chats through the Rest API it lays out. The chat User Interface(UI), is created from the NPM components this tool comes with. See how the hustle is waved away for us already?
+Also, thanks to its server(s), the chat can be hosted from here(the server).
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:26]: Format your paragraphs well..

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+
+### Creating the Application
+This minute, let's begin.
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:30]: Begin doing what? Do not use incomplete sentences..

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+- Head to your windows terminal and navigate to a new directory named to your preference, and create a  new react app. You could opt to use the Vite tool instead whose step-by-step guide is taught in [this](https://www.section.io/engineering-education/creating-a-react-app-using-vite/) article. 
+  Else, run the command below with the project name replaced with an application name of your choice. You may opt to use ***yarn create-react-app app-name***.
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:32]: Simplify your sentences, and do not let users guess of what should happen, give detailed description. For example, say ` On your terminal, cd to project directory and run the following commands`. be precise.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+  Save the Project ID and the secret keys somewhere safe. We will use them along the way.
+
+- Still, on this page, click the "new user" section at the top right corner and give a name to your new  user in a modal that pops up.
+  ![New User](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/user.png)
+
+> You may need to create a users' list to select from, when starting a direct chat later.
+
+As a developer, you may want to always practise the separation of **sensitive** information like your userSecret, you obtained and the projectID from the rest of the code to avoid such information being exposed to **unauthorized users**, hence protecting your data. 
+Therefore, create a .env file at the root of your directory and have the ***userSecret*** and ***projectID*** saved as environment variables.
+Remember to add the **.env** file to the **.gitignore** file to avoid pushing such sensitive data to Github.
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:62]: You have used `userSecret`, among others, kindly rewrite them well. such as ` user secret'.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+![Selecting user](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/select.png)
+
+- Type on the text input field and click the button labeled ***"send"*** 
+![Start chat](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/chat.png)
+
+You can keep texting the other user however much you please. Observe, the left section with the chat title, displays the last message you send. 
+![Final App](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/finalapp.png)
+
+Getting your hands dirty with code couldn't be more fun! Right?
+
```
Comment (owinowendy) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:223]: Include link to this project source code.

**Comment by Neema-2016** [comment]
Hey, @owinowendy I have made all the requested changes, kindly check. I however was not sure about the spacing correction I was to make on the title. Thank you.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
+---
+title: Creating a Simple Chat App in React.js using the chat engine API
+---
+
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:4]: @Neema-2016, kindly remove the `---` tags and just have standard heading format.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+This article will teach the React.js way of creating a chat application using the chat engine API. It aims at making it less cumbersome compared to when one opts the Node.js and socket.io way instead.
+
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:6]: This first sentence on you introduction is quite confusing. Why are you comparing Node.js and Socket.io? What if readers have no prior knowledge?

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+The learner will understand how the chat engine API works and how to use it to create a precise yet functional direct-messaging application.
+
+
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:9]: Remove/minimis this spacing..and anywhere applicable.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+Approximating to other APIs that deliver user responses to the system, and sends back the system's responses to the user, this API works the same way, with better functionalities.
+
+This tool saves you a lot of implementation time otherwise needed on many lines of code.
+Chat engine API can implement; 
+ - A group chat application.
+ - Direct messages chat application that would work like Facebook messenger .
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:15]: An introduction is meant to inform reader what to expect, I feel you should remove these paragraphs from your introduction and add them to the main body where applicable.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+
+### Prerequisites
+You will need to be equipped with what follows next before jumping right to it.
+1. Primary React.js library mastery. Head to [the official documentation](https://reactjs.org/tutorial/tutorial.html) if there's a need to familiarize yourself with React.js.
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:19]: Do not make this sentence long, simplify. For example, `1.[Node.js](link).` While you're informing reader of the knowledge required, you're at the same time giving them links to refer to.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+2. Installed on your system Node.js, which you could access from [here](https://nodejs.org/en/download/). 
+3. [Node Package Manager (NPM)](https://docs.npmjs.com/cli/v6/commands/npm) know-how.
+
+### What chat engine API is?
+The chat engine is an API consumed to build chat services. It allows the hosting of these chats through the Rest API it lays out. The chat User Interface(UI), is created from the NPM components this tool comes with. See how the hustle is waved away for us already? 
+Furthermore, thanks to its server(s), the chat can as well be hosted from here(the server).
+
+
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:27]: Work on the spacings....

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+### Creating the Application
+This minute, let's begin creating the application.
+- On the terminal, cd to a new directory named to your preference and run the following commands. If you prefer using the Vite tool to create the React app, its step-by-step guide is taught in [this](https://www.section.io/engineering-education/creating-a-react-app-using-vite/) article. 
+  Else, run the command below with the project name replaced with an application name of your choice. You may opt to use ***yarn create-react-app app-name***.
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:31]: This was pointed our previously and you have not fixed it.

**Comment by Neema-2016** [comment]
Hey, @jumamiller I just made the changes requested, please check. My bad I failed to make every previously requested changes.

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 18]
```diff
@@ -0,0 +1,208 @@
+title: Creating a Simple Chat App in React.js using the chat engine API
+
+This article will teach the React.js way of creating a chat application using the chat engine API. 
+The learner will understand how the chat engine API works and how to use it to create a precise yet functional direct-messaging application.
+
+### Prerequisites
+1. Primary React.js library mastery.  [The official documentation](https://reactjs.org/tutorial/tutorial.html) 
+2. [Node.js](https://nodejs.org/en/download/) installed. 
+3. [Node Package Manager (NPM)](https://docs.npmjs.com/cli/v6/commands/npm) know-how.
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md:9]: Use dash instead of 1,2,3 numbering system, for example, `- task 1`

**Review by jumamiller — CHANGES_REQUESTED** [review]
Fix above suggestions before we move to technical review...

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 19]
```diff
@@ -0,0 +1,208 @@
...
+### What chat engine API is?
+The chat engine is an API consumed to build chat services. 
+Approximating to other APIs that deliver user responses to the system, and sends back the system's responses to the user, this API works the same way, with better functionalities.
+It allows the hosting of these chats through the Rest API it lays out. The chat User Interface(UI), is created from the NPM components this tool comes with. See how the hustle is waved away for us already? 
+Furthermore, thanks to its server(s), the chat can as well be hosted from here(the server).
+This tool also saves you a lot of implementation time otherwise needed on many lines of code.
+Chat engine API can implement; 
+ - A group chat application.
+ - Direct messages chat application that would work like Facebook messenger .
+
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md]: This section here is quite hard to follow. I wanted to get more info about the Chat API, but I couldn't understand anything about it. Kindly fix all these, I need to follow each step while working on tech review

File: content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+### Creating the Application
+This minute, let's begin creating the application.
+- On the terminal, cd to your project directory and run the following commands. 
+
+```bash
+npx create-react-app project-name
+```
+
+If you prefer using the Vite tool to create the React app, its step-by-step guide is taught in [this](https://www.section.io/engineering-education/creating-a-react-app-using-vite/) article.
```
Comment (jumamiller) [inline content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md]: What's Vite? some readers are just curious to read anything on the internet, let them understand everything you're discussing.

**Comment by Kayere** [comment]
Great article @Neema-2016.

Thank you for the review @jumamiller.
